File: matchmaker/app/src/test/java/ru/ifmo/se/dating/matchmaker/DatabaseInitializer.java
package ru.ifmo.se.dating.matchmaker;

import org.springframework.boot.test.util.TestPropertyValues;
import org.springframework.context.ApplicationContextInitializer;
import org.springframework.context.ConfigurableApplicationContext;

import lombok.ToString;
import ru.ifmo.se.dating.container.Postgres;

@ToString
public final class DatabaseInitializer
    implements ApplicationContextInitializer<ConfigurableApplicationContext> {

    private final Postgres postgres;

    DatabaseInitializer() {
        this.postgres = Postgres.start();
    }

    @Override
    public void initialize(ConfigurableApplicationContext applicationContext) {
        TestPropertyValues.of(
                "spring.datasource.url=%s".formatted(postgres.jdbcUrl()),
                "spring.datasource.username=%s".formatted(postgres.username()),
                "spring.datasource.password=%s".formatted(postgres.password()))
                .applyTo(applicationContext.getEnvironment());
    }
}

File: backend/matchmaker/src/test/kotlin/ru/ifmo/se/dating/matchmaker/DatabaseInitializer.kt
package ru.ifmo.se.dating.matchmaker

import org.springframework.boot.test.util.TestPropertyValues
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.ConfigurableApplicationContext
import ru.ifmo.se.dating.container.Postgres

class DatabaseInitializer :
    ApplicationContextInitializer<ConfigurableApplicationContext> {
    private val postgres: Postgres = Postgres.start()

    override fun initialize(ctx: ConfigurableApplicationContext) {
        TestPropertyValues.of(
            "spring.datasource.url=${postgres.jdbcUrl()}",
            "spring.datasource.username=${postgres.username()}",
            "spring.datasource.password=${postgres.password()}",
        ).applyTo(ctx.environment)
    }
}
